File: src/types/prompt.ts
export interface Prompt {
  id: string;
  title: string;
  content: string;      // 命令或 Prompt 模板
  group: string;        // 所属分组ID或名称
  description?: string; // 简短描述
  tags?: string[];      // 标签
  isFavorite: boolean;
  createdAt: number;
  updatedAt: number;
  
  // ✨ 新增字段
  source?: 'local' | 'official'; // 来源
  packId?: string;               // 所属包ID (仅 official 有)
  originalId?: string;           // ✨ 关键修复：记录克隆自哪个官方指令 ID
}

export const DEFAULT_GROUP = 'Default';

// 商店清单数据结构
export interface PackManifestItem {
  id: string;        // e.g. "zh-linux"
  language: string;  // "zh"
  platform: string;  // "linux"
  name: string;      // "Linux 运维"
  description: string;
  count: number;
  size_kb: number;
  url: string;       // 相对路径 "packs/zh/linux.json"
}

export interface PackManifest {
  updated_at: number;
  version: string;
  packages: PackManifestItem[];
}

File: src-tauri/crates/ctxrun-model/src/prompt.rs
use serde::{Deserialize, Serialize};
use specta::Type;

// ============================================================================
// Data Models
// ============================================================================

#[derive(Debug, Serialize, Deserialize, Clone, Type)]
#[serde(rename_all = "camelCase")]
pub struct Prompt {
    pub id: String,
    pub title: String,
    pub content: String,
    #[serde(rename = "group")]
    pub group_name: String,
    pub description: Option<String>,
    pub tags: Option<Vec<String>>,
    pub is_favorite: bool,
    pub created_at: i64,
    pub updated_at: i64,
    pub source: String,
    pub pack_id: Option<String>,
    pub original_id: Option<String>,
    #[serde(rename = "type")]
    pub type_: Option<String>,
    pub is_executable: Option<bool>,
    pub shell_type: Option<String>,
    pub use_as_chat_template: Option<bool>,
}

#[derive(serde::Serialize, Type)]
pub struct PromptCounts {
    pub prompt: i64,
    pub command: i64,
}

// ============================================================================
// CSV Export/Import Models
// ============================================================================

#[derive(Debug, Serialize, Deserialize, Type)]
#[allow(dead_code)]
pub struct PromptCsvRow {
    #[serde(default)]
    pub id: Option<String>,
    pub title: String,
    pub content: String,
    #[serde(rename = "group")]
    pub group_name: String,
    pub description: Option<String>,
    #[serde(default)]
    pub tags: String,
    #[serde(default)]
    pub is_favorite: bool,
    #[serde(rename = "type", default = "default_type")]
    pub type_: String,
    #[serde(default)]
    pub is_executable: bool,
    pub shell_type: Option<String>,
}

#[allow(dead_code)]
fn default_type() -> String {
    "prompt".to_string()
}
